bliss/hyperparameter.py

```diff
@@ -0,0 +1,315 @@
+
+        # set up the checkpoint callback for the model
+        checkpoint_callback = pl.callbacks.ModelCheckpoint(
+            os.path.join(self.model_dir, "trial_{}".format(trial.number), "{epoch}"),
+            monitor=self.monitor,
+        )
+
+        # Initiate the model
+        model = SleepPhase(self.cfg)
+
+        # put correct device to model
+        use_gpu = 0
+        if self.gpu_queue is not None and self.single_gpu is None:
+            use_gpu = [gpu_id]
+        elif self.gpu_queue is None and self.single_gpu is not None:
+            use_gpu = [self.single_gpu]
+
+        # put gpu id back to the queue
+        if self.gpu_queue is not None:
+            self.gpu_queue.put(gpu_id)
+
+        # set up the trainer
+        ## PyTorchLightningPruningCallback allow the pruner to stop early
+        trainer = pl.Trainer(
+            logger=False,
+            gpus=use_gpu,
+            checkpoint_callback=checkpoint_callback,
+            max_epochs=self.max_epochs,
+            callbacks=[
+                self.metrics_callback,
+                PyTorchLightningPruningCallback(trial, monitor=self.monitor),
+            ],
+        )
+
+        # start training
+        trainer.fit(model, datamodule=star_dataset)
+
+        return self.metrics_callback.metrics[-1][self.monitor].item()
+
+
+def SleepTune(
+    cfg: DictConfig,
+    max_epochs: int,
+    model_dir,
+    monitor,
+    n_batches,
+    batch_size,
+    direction: str = "minimize",
+    data_seed: int = 10,
+    n_trials=100,
+    time_out=600,
+    gc_after_trial: bool = False,
+    sampler=None,
+    pruner=None,
+    num_gpu: int or list = 2,
+    storage=None,
+):
+    """
+    :param cfg: DictConfig from hydra
+    :param max_epochs: in integer to indicate epochs of training
+    :param model_dir: where to save the model checkpoint of lightning
+    :param monitor: which metric to monitor for the hyperparameter selection
+    :param n_batches: how many batches per epoch
+    :param batch_size: the size of image simulated for each batch
+    :param direction: minimize or maximize the chosen metric
+    :param data_seed: set a seed for data simulation
+    :param n_trials: the maximum trials to run
+    :param time_out: the maximum time to run for each trial
+    :param gc_after_trial: enable garbage collection after each trial?
+    :param sampler: optuna.sampler
+    :param pruner: optuna.pruner
+    :param num_gpu: numer of GPUs to use. Could be an integer or a list of ids.
+                    CPU only: set this arugument to be zero
+                    Single GPU: either pass in 1 ot a list single id, e.g. [1]
+                    Multiple GPUs: either pass in an integer > 1 and SleepTune pick
+                                    available GPUS, or a list of ids.
+    :param storage: If using multiple GPUs, storage must be defined as optuna.RDBStorage
+
+    :return: an optuna.trial.FrozenTrial object
+    """
```

Comment:
Could you change the format of the docstring to the google format? Thanks! (there is an option in pycharm settings)